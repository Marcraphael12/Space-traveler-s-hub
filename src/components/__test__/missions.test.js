import TestRenderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';
import { Provider } from 'react-redux';
import store from '../../redux/configureStore';
import Missions from '../missions/missions';

describe('Display the missions list', () => {
  it('Renders correctly every missions from the API', () => {
    const mission = {
      id: 2,
      mission_name: 'one',
      description: 'a mission',
      joined: true,
    };
    const missions = TestRenderer.create(
      <Provider store={store}>
        <Missions key={mission.id} missions={mission} />
      </Provider>,
    ).toJSON();
    expect(missions).toMatchSnapshot();
  });
});