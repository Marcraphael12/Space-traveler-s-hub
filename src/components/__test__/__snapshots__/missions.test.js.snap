// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` Renders correctly every missions from the API 1`] = `
<div
  className="container"
>
  <table
    className="table table-bordered table-striped"
  >
    <thead>
      <tr>
        <th
          scope="col"
          style={
            Object {
              "width": "10%",
            }
          }
        >
          Mission
        </th>
        <th
          scope="col"
          style={
            Object {
              "width": "60%",
            }
          }
        >
          Description 
        </th>
        <th
          scope="col"
          style={
            Object {
              "width": "15%",
            }
          }
        >
          Status
        </th>
        <th
          scope="col"
          style={
            Object {
              "width": "15%",
            }
          }
        >
          Join
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;

exports[`Display the missi Renders correctly every missions from the API 1`] = `
<div
  className="container"
>
  <table
    className="table table-bordered table-striped"
  >
    <thead>
      <tr>
        <th
          scope="col"
          style={
            Object {
              "width": "10%",
            }
          }
        >
          Mission
        </th>
        <th
          scope="col"
          style={
            Object {
              "width": "60%",
            }
          }
        >
          Description 
        </th>
        <th
          scope="col"
          style={
            Object {
              "width": "15%",
            }
          }
        >
          Status
        </th>
        <th
          scope="col"
          style={
            Object {
              "width": "15%",
            }
          }
        >
          Join
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;

exports[`Display the missions list Renders correctly every missions from the API 1`] = `
<div
  className="container"
>
  <table
    className="table table-bordered table-striped"
  >
    <thead>
      <tr>
        <th
          scope="col"
          style={
            Object {
              "width": "10%",
            }
          }
        >
          Mission
        </th>
        <th
          scope="col"
          style={
            Object {
              "width": "60%",
            }
          }
        >
          Description 
        </th>
        <th
          scope="col"
          style={
            Object {
              "width": "15%",
            }
          }
        >
          Status
        </th>
        <th
          scope="col"
          style={
            Object {
              "width": "15%",
            }
          }
        >
          Join
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;

exports[`Render the missions lists Renders correctly every missions from the  1`] = `
<div
  className="container"
>
  <table
    className="table table-bordered table-striped"
  >
    <thead>
      <tr>
        <th
          scope="col"
          style={
            Object {
              "width": "10%",
            }
          }
        >
          Mission
        </th>
        <th
          scope="col"
          style={
            Object {
              "width": "60%",
            }
          }
        >
          Description 
        </th>
        <th
          scope="col"
          style={
            Object {
              "width": "15%",
            }
          }
        >
          Status
        </th>
        <th
          scope="col"
          style={
            Object {
              "width": "15%",
            }
          }
        >
          Join
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;

exports[`Render the missions lists Renders the missions correctly 1`] = `
<div
  className="container"
>
  <table
    className="table table-bordered table-striped"
  >
    <thead>
      <tr>
        <th
          scope="col"
          style={
            Object {
              "width": "10%",
            }
          }
        >
          Mission
        </th>
        <th
          scope="col"
          style={
            Object {
              "width": "60%",
            }
          }
        >
          Description 
        </th>
        <th
          scope="col"
          style={
            Object {
              "width": "15%",
            }
          }
        >
          Status
        </th>
        <th
          scope="col"
          style={
            Object {
              "width": "15%",
            }
          }
        >
          Join
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;

exports[`Render the missions lists Renders y 1`] = `
<div
  className="container"
>
  <table
    className="table table-bordered table-striped"
  >
    <thead>
      <tr>
        <th
          scope="col"
          style={
            Object {
              "width": "10%",
            }
          }
        >
          Mission
        </th>
        <th
          scope="col"
          style={
            Object {
              "width": "60%",
            }
          }
        >
          Description 
        </th>
        <th
          scope="col"
          style={
            Object {
              "width": "15%",
            }
          }
        >
          Status
        </th>
        <th
          scope="col"
          style={
            Object {
              "width": "15%",
            }
          }
        >
          Join
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;
